### DISCUSSION TITLE
Help me to understand the "Disk cache size" option?

### DISCUSSION THREAD

**Discussion by aqsede** [transmission/transmission#7127] [Q&A]
Hello, 

By default disk cache size is set to 4MB.
I have a lot of access RAM (idle is using 3GB/32GB) 

and now I set the size to 4096MB.

I have this hypothetical situation:
- 100s of torrents with 10 different file around 2GB each inside. (Easier to assume this is tv series)
- Seeding is slow but always active at maybe 10KB to 100KB

that I'm trying to accomplish:
- Transmission doesn't wake disk but put in cache intelligently for this slow seeding and read from there until the cache need to be updated

Is this correct or I'm trying to do something that is still not yet feasible?

Thanks!

**Comment by jfeilbach** [comment]
There is no option called "disk cache size" There is only `"cache-size-mb": 4096,` which has a default value of 4 MegaBytes. This specifies the amount of memory to use to cache data. Increasing this value will cause transmission to use more memory (RAM). You need to set this to an integer only like `4096` not `4096MB` meaning you would use ~4 GB of memory as disk cache. In your example where you have roughly 2 TB of data you will need a significant amount of memory in order to cache all of that which you don't appear to have. The advantage of this setting is when data is accessed more than once resulting in a cache hit and not needing to access disk. This value is a trade off with the OS and other applications to reduce disk usage. Assuming your operating system is Linux you should learn to understand how memory is used and the values stored in `/proc/meminfo` as read by the command `free`. 

```
$ free -h
               total        used        free      shared  buff/cache   available
Mem:            31Gi       631Mi       583Mi       1.0Mi        30Gi        30Gi
Swap:          8.0Gi       126Mi       7.9Gi
```

This might be a good place to start to learn about memory values https://www.redhat.com/sysadmin/dissecting-free-command
To better understand how a HDD sleeps in Linux `man hdparm` Specifically you are interested in this setting `Advanced power management level`
